vms/platformvm/validators/manager_benchmark_test.go

```diff
@@ -0,0 +1,268 @@
+	genesisValidators := []api.PermissionlessValidator{{
+		Staker: api.Staker{
+			StartTime: json.Uint64(genesisTime.Unix()),
+			EndTime:   json.Uint64(genesisEndTime.Unix()),
+			NodeID:    ids.GenerateTestNodeID(),
+		},
+		RewardOwner: &api.Owner{
+			Threshold: 1,
+			Addresses: []string{addr},
+		},
+		Staked: []api.UTXO{{
+			Amount:  json.Uint64(2 * units.KiloAvax),
+			Address: addr,
+		}},
+		DelegationFee: reward.PercentDenominator,
+	}}
+
+	buildGenesisArgs := api.BuildGenesisArgs{
+		NetworkID:     json.Uint32(constants.UnitTestID),
+		AvaxAssetID:   avaxAssetID,
+		UTXOs:         nil,
+		Validators:    genesisValidators,
+		Chains:        nil,
+		Time:          json.Uint64(genesisTime.Unix()),
+		InitialSupply: json.Uint64(360 * units.MegaAvax),
+		Encoding:      formatting.Hex,
+	}
+
+	buildGenesisResponse := api.BuildGenesisReply{}
+	platformvmSS := api.StaticService{}
+	require.NoError(platformvmSS.BuildGenesis(nil, &buildGenesisArgs, &buildGenesisResponse))
+
+	genesisBytes, err := formatting.Decode(buildGenesisResponse.Encoding, buildGenesisResponse.Bytes)
+	require.NoError(err)
+
+	vdrs := validators.NewManager()
+	vdrs.Add(constants.PrimaryNetworkID, validators.NewSet())
+
+	execConfig, err := config.GetExecutionConfig(nil)
+	require.NoError(err)
+
+	metrics, err := metrics.New("", prometheus.NewRegistry())
+	require.NoError(err)
+
+	s, err := state.New(
+		db,
+		genesisBytes,
+		prometheus.NewRegistry(),
+		&config.Config{
+			Validators: vdrs,
+		},
+		execConfig,
+		&snow.Context{
+			NetworkID: constants.UnitTestID,
+			NodeID:    ids.GenerateTestNodeID(),
+			Log:       logging.NoLog{},
+		},
+		metrics,
+		reward.NewCalculator(reward.Config{
+			MaxConsumptionRate: .12 * reward.PercentDenominator,
+			MinConsumptionRate: .10 * reward.PercentDenominator,
+			MintingPeriod:      365 * 24 * time.Hour,
+			SupplyCap:          720 * units.MegaAvax,
+		}),
+		new(utils.Atomic[bool]),
+	)
+	require.NoError(err)
+
+	m := NewManager(
+		logging.NoLog{},
+		config.Config{
+			Validators: vdrs,
+		},
+		s,
+		metrics,
+		new(mockable.Clock),
+	)
+
+	var (
+		nodeIDs       []ids.NodeID
+		currentHeight uint64
+	)
+	for i := 0; i < 50; i++ {
+		require.NoError(addPrimaryValidator(s, genesisTime, genesisEndTime, &nodeIDs, &currentHeight))
+	}
+	subnetID := ids.GenerateTestID()
+	for _, nodeID := range nodeIDs {
+		require.NoError(addSubnetValidator(s, subnetID, genesisTime, genesisEndTime, nodeID, &currentHeight))
+	}
+	for i := 0; i < 9900; i++ {
+		require.NoError(addSubnetDelegator(s, subnetID, genesisTime, genesisEndTime, nodeIDs, &currentHeight))
+	}
+
+	ctx := context.Background()
+	height, err := m.GetCurrentHeight(ctx)
+	require.NoError(err)
+	require.Equal(currentHeight, height)
+
+	b.ResetTimer()
+
+	for i := 0; i < b.N; i++ {
+		_, err := m.GetValidatorSet(ctx, 0, subnetID)
+		require.NoError(err)
+	}
+}
+
+func addPrimaryValidator(s state.State, startTime, endTime time.Time, nodeIDs *[]ids.NodeID, height *uint64) error {
```

Comment:
Would it be cleaner to return the newly added nodeID rather than taking in a pointer to `nodeIDs` so that this function can mutate it?